### DISCUSSION TITLE
Unique constraint and Index SQL script keyword support in EntityMappingBuilder<T>

### DISCUSSION THREAD

**Discussion by Arkatufus** [linq2db/linq2db#3972] [Q&A] [answered]
Hi,

I was wondering if I can express unique constraint and column indexing on table columns using the `EntityMappingBuilder<T>` and `CreateTable<T>()` method in Linq2Db. Is this something that Linq2Db supports and if it is, is there a sample code that I can see?

**Comment by MaceWindu** [comment, marked as answer]
This is not supported currently. How you expect linq2db to use this information?

**Reply by Arkatufus** [reply]
I'm trying to leverage the `CreateTable<T>()` method to automatically populate the database with the appropriate table before the application starts, the unique constraint and column indexing was needed to make sure that row insertion is correct and the table is optimized for read operation. Thank you for the answer.

**Reply by MaceWindu** [reply]
Yes, currently we don't provide functionality to create indexes and other constrains except primary key. We could add it at some point but it's not in our priority list. But we open for PR
